File: Test/NL_TEST/test1.js
// 개미수열

function solution(n) {
    let answer = "";
    if (n === 1) return "1";
    answer = go(solution(n - 1));
    return answer;
}


function go(str) {
    let arr1 = [];
    let arr2 = [];
    let result = "";

    while (str.length) {
        if (str[0] === str[1]) {
            arr2.push(str[0]);
            str = str.replace(str[0], "");
        } else {
            arr2.push(str[0]);
            str = str.replace(str[0], "");
            arr1.push(arr2);
            arr2 = [];
        }
    }
    arr1.forEach(a => {
        result += `${a[0]}${a.length}`;
    })
    return result;
}

File: Test/21라상_TEST/test1.py
def solution(table, languages, preference):
    answer = ''
    my_dict = {}
    result = {}
    for item in table:
        [job, l1, l2, l3, l4, l5] = item.split()
        my_dict[job] = [0, l5, l4, l3, l2, l1]

    for i in range(len(languages)):
        prefer = preference[i]
        for job in my_dict:
            if languages[i] in my_dict[job]:
                score = my_dict[job].index(languages[i])
                if result.get(job) == None:
                    result[job] = int(score * prefer)
                else:
                    result[job] += int(score * prefer)
    sorted_result = sorted(result.items(), key=lambda x: (-x[1], x[0]))
    answer = sorted_result[0][0]
    return answer
